jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPart.java

```diff
@@ -347,15 +361,18 @@ public String toString()
      */
     public static class ChunksPart extends Part
     {
-        private final List<Content.Chunk> content;
-        private final List<Content.Source> contentSources = new ArrayList<>();
         private final AutoLock lock = new AutoLock();
+        private final List<Content.Chunk> content = new ArrayList<>();
+        private final List<Content.Source> contentSources = new ArrayList<>();
         private boolean closed = false;
 
         public ChunksPart(String name, String fileName, HttpFields fields, List<Content.Chunk> content)
         {
             super(name, fileName, fields);
-            this.content = Objects.requireNonNull(content);
+            try (AutoLock l = lock.lock())
```

Comment:
Don't think it's necessary to lock here (because the constructor publishes)?

Reply:
True, a barrier is emitted at the end of the constructor because there are final fields. Let me remove this lock.